api/client/src/main/java/org/projectnessie/client/rest/v2/HttpGetDiff.java

```diff
@@ -48,27 +51,33 @@ protected DiffParams params() {
 
   @Override
   public DiffResponse get(DiffParams params) throws NessieNotFoundException {
-    HttpRequest req =
-        client
-            .newRequest()
-            .path("trees/{from}/diff/{to}")
-            .resolveTemplate("from", params.getFromRef())
-            .resolveTemplate("to", params.getToRef())
-            .queryParam("max-records", params.maxRecords())
-            .queryParam("page-token", params.pageToken())
-            .queryParam("filter", params.getFilter());
-    params.getRequestedKeys().forEach(k -> req.queryParam("key", k.toPathString()));
+    HttpRequest req = client.newRequest();
+
+    if (api.isNessieSpec220()) {
+      return req.path("trees/.diff")
+          .unwrap(NessieNotFoundException.class)
+          .post(ImmutableDiffJson.builder().from(params).build())
```

Comment:
I'm not sure about checking the spec version for choosing request methods is in line with how the spec version was envisioned to be used... The spec version defines server behaviour, so it's file to use it as a way to determine whether the server is able to parse the new `ContentKey` encoding, but as far as REST API goes, I think we have to bump the API version for new methods (even if it's backward compatible).